src/services/car/inventory.ts

```diff
@@ -286,6 +286,36 @@ export const transferDealerListingToManual = async ({
   }
 }
 
+export const transferDealerListingsToManual = async ({
+  dealerId,
+  listingIds,
+  options = {},
+}: {
+  dealerId: number
+  listingIds: number[]
+  options: ApiCallOptions
+}): Promise<WithValidationError> => {
+  try {
+    await postData({
+      path: `/api/dealers/${dealerId}/listings/transfer-to-manual`,
+      body: {
+        elements: listingIds,
+      },
+      options: {
+        isAuthorizedRequest: true,
+        ...options,
+      },
+    })
+  } catch (error) {
+    return handleValidationError(error)
+  }
+
+  return {
```

Comment:
Wont this potentially return before the promise is fullfilled so in case of an error we might get the wrong return value?

Reply:
No, there's an `await` in line [209](https://github.com/carforyou/carforyou-api-client-pkg/pull/237/files#diff-07e17215dafd3af97dcd5add6412634ccd2600a5bd5f046a56a8ffa81fef73c6R299)

Reply:
yes but the await is inside the try catch block, https://playcode.io/745138/

Reply:
`setTimeout` behaves differently than promises. Also in the linked playground, you're not awaiting for the timed out function to finish execution (so it's similar to an example that would omit `await`).
Here's the example that works closer to what you want to test: https://playcode.io/745139/

It's the pattern we use already in the project and we have good test coverage here, I'm sure we would have caught an issue if it was one.

Reply:
nvm, I messed something up in my head, thanks for checking

Reply:
No worries. it's good to check things and ask these questions. There was a risk that something was not right there 👍